src/lib/zkapp.ts

```diff
@@ -1422,15 +1422,15 @@ Use the optional \`maxTransactionsWithActions\` argument to increase this number
       fromActionState: Field,
       config
     ): Field {
-      const stateType = provable(undefined);
+      const stateType = provable(null);
```

Comment:
yeah, so `state = undefined` was causing this line to fail:

```ts
Provable.assertEqual(stateType, newState, state);
```

Because how `assertEqual` is implemented, it looks whether the last argument is undefined to determine its behaviour:
https://github.com/o1-labs/o1js/blob/3660fe37aa3d32cc788704962556fc780d1879c3/src/lib/provable.ts#L244-L250

I wonder if `assertEqual` could behave more intelligently 🤔

Reply:
Yeah probably `assertEqual` should also check for the existence of a `toFields()` method on `xOrY` as a second hint before taking the "implicit" path

Reply:
or it could check whether the `typeOrX` matches some known `Undefined` provable type, in which case it takes the explicit path

Reply:
these overloaded APIs are a bit messy..